install.sh

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env bash
+# Copyright 2024 Science project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+set -euo pipefail
+
+COLOR_RED="\x1b[31m"
+COLOR_GREEN="\x1b[32m"
+COLOR_YELLOW="\x1b[33m"
+COLOR_RESET="\x1b[0m"
+
+function log() {
+  echo -e "$@" 1>&2
+}
+
+function die() {
+  (($# > 0)) && log "${COLOR_RED}$*${COLOR_RESET}"
+  exit 1
+}
+
+function green() {
+  (($# > 0)) && log "${COLOR_GREEN}$*${COLOR_RESET}"
+}
+
+function warn() {
+  (($# > 0)) && log "${COLOR_YELLOW}$*${COLOR_RESET}"
+}
+
+function ensure_cmd() {
+  local cmd="$1"
+  command -v "$cmd" > /dev/null || die "This script requires the ${cmd} binary to be on \$PATH."
+}
+
+ISSUES_URL="https://github.com/a-scie/lift/issues"
+_GC=()
+
+ensure_cmd rm
+function gc() {
+  if (($# > 0)); then
+    _GC+=("$@")
+  else
+    # Check if $_GC has members to avoid "unbound variable" warnings if gc w/ arguments is never called.
+    if ! [ ${#_GC[@]} -eq 0 ]; then
+      rm -rf "${_GC[@]}"
+    fi
+  fi
+}
+
+trap gc EXIT
+
+ensure_cmd uname
+function determine_os() {
+  local os
+
+  os="$(uname -s)"
+  if [[ "${os}" =~ [Ll]inux ]]; then
+    echo linux
+  elif [[ "${os}" =~ [Dd]arwin ]]; then
+    echo macos
+  elif [[ "${os}" =~ [Ww]indow|[Mm][Ii][Nn][Gg] ]]; then
+    # Powershell reports something like: Windows_NT
+    # Git bash reports something like: MINGW64_NT-10.0-22621
+    echo windows
+  else
+    die "Science is not supported on this OS (${os}). Please reach out at ${ISSUES_URL} for help."
+  fi
+}
+
+OS="$(determine_os)"
+
+ensure_cmd arch
```

Comment:
No need for an extra dep, `arch` == `uname -m` and you use `uname -s` above.

Reply:
sgtm